scripts/src/javatests/org/oppia/android/scripts/license/MavenDependenciesRetrieverTest.kt

```diff
@@ -0,0 +1,1312 @@
+        this.artifactVersion = PROTO_LITE_VERSION
+      }.build(),
+      MavenDependency.newBuilder().apply {
+        this.artifactName = DEP_WITH_INVALID_LINKS
+        this.artifactVersion = IO_FABRIC_VERSION
+        this.addLicense(license2)
+      }.build()
+    )
+
+    val depsThatNeedInterventionSet =
+      MavenDependenciesRetriever.getDependenciesThatNeedIntervention(mavenDependenciesList)
+    assertThat(depsThatNeedInterventionSet.size).isEqualTo(2)
+    assertIsDependency(
+      dependency = depsThatNeedInterventionSet.elementAt(0),
+      artifactName = DEP_WITH_NO_LICENSE,
+      artifactVersion = PROTO_LITE_VERSION
+    )
+    assertIsDependency(
+      dependency = depsThatNeedInterventionSet.elementAt(1),
+      artifactName = DEP_WITH_INVALID_LINKS,
+      artifactVersion = IO_FABRIC_VERSION
+    )
+  }
+
+  @Test
+  fun testRetrieveMavenDepList_emptyPbFile_returnsEmptyDepsList() {
+    val pbFile = tempFolder.newFile("scripts/assets/maven_dependencies.pb")
+    val mavenDependencyList = MavenDependencyList.newBuilder().build()
+
+    mavenDependencyList.writeTo(pbFile.outputStream())
+
+    val mavenDependenciesList = MavenDependenciesRetriever.retrieveMavenDependencyList(
+      "${tempFolder.root}/scripts/assets/maven_dependencies.pb"
+    )
+
+    assertThat(mavenDependenciesList).isEmpty()
+  }
+
+  @Test
+  fun testRetrieveMavenDepList_nonEmptyMavenDependencyList_returnsCorrectDepsList() {
+    val pbFile = tempFolder.newFile("scripts/assets/maven_dependencies.pb")
+
+    val license1 = License.newBuilder().apply {
+      this.licenseName = "The Apache License, Version 2.0"
+      this.originalLink = "https://www.apache.org/licenses/LICENSE-2.0.txt"
+      this.scrapableLink = ScrapableLink.newBuilder().apply {
+        url = "https://www.apache.org/licenses/LICENSE-2.0.txt"
+      }.build()
+    }.build()
+    val license2 = License.newBuilder().apply {
+      this.licenseName = "Simplified BSD License"
+      this.originalLink = "https://www.opensource.org/licenses/bsd-license"
+    }.build()
+    val mavenDependencyList = MavenDependencyList.newBuilder().apply {
+      this.addAllMavenDependency(
+        listOf(
+          MavenDependency.newBuilder().apply {
+            this.artifactName = DEP_WITH_DIRECT_LINK_ONLY_LICENSE
+            this.artifactVersion = FIREBASE_ANALYTICS_VERSION
+            this.addLicense(license1)
+          }.build(),
+          MavenDependency.newBuilder().apply {
+            this.artifactName = DEP_WITH_NO_LICENSE
+            this.artifactVersion = PROTO_LITE_VERSION
+            this.addLicense(license2)
+          }.build()
+        )
+      )
+    }.build()
+
+    mavenDependencyList.writeTo(pbFile.outputStream())
+
+    val mavenDependenciesList = MavenDependenciesRetriever.retrieveMavenDependencyList(
+      "${tempFolder.root}/scripts/assets/maven_dependencies.pb"
+    )
+
+    assertThat(mavenDependenciesList.size).isEqualTo(2)
```

Comment:
Could you just check that the two protos are equal, instead? I think Truth comes with a proto subject, too: https://truth.dev/api/0.43/com/google/common/truth/extensions/proto/LiteProtoSubject.html.

Ditto elsewhere.

Reply:
Actually, I am verifying each and every field explicitly as per your comment here - https://github.com/oppia/oppia-android/pull/3434#discussion_r671428912